client/product-details/index.js

```diff
@@ -7,51 +7,95 @@
-	const { productVariations } = window.wcpayStripeSiteMessaging;
-	let { productId } = window.wcpayStripeSiteMessaging;
+
+	// Utility function to safely parse integers and handle NaN
+	const safeParseInt = ( value ) => {
+		const result = parseInt( value, 10 );
+		return isNaN( result ) ? 0 : result;
+	};
+
+	const updateBnplPaymentMessage = ( amount, currency ) => {
+		if ( amount > 0 && currency ) {
+			bnplPaymentMessageElement.update( { amount, currency } );
+		}
+	};
+
+	const quantityInput = $( '.quantity input[type=number]' );
 
 	const resetBnplPaymentMessage = () => {
-		const quantity = $( '.quantity input[type=number]' ).val();
-		productId = 'base_product';
-		bnplPaymentMessageElement.update( {
-			amount:
-				parseInt( productVariations.base_product.amount, 10 ) *
-				quantity,
-			currency: productVariations.base_product.currency,
-		} );
+		if ( ! quantityInput.length || ! productVariations.base_product )
+			return;
+
+		const quantity = safeParseInt( quantityInput.val() );
+		const baseProductAmount = safeParseInt(
+			productVariations.base_product.amount
+		);
+
+		const amount = baseProductAmount * quantity;
+		const currency = productVariations.base_product.currency;
+
+		updateBnplPaymentMessage( amount, currency );
 	};
 
-	$( '.quantity input[type=number]' ).on( 'change', function ( event ) {
-		const newQuantity = event.target.value;
-		const price = productVariations[ productId ].amount;
-		bnplPaymentMessageElement.update( {
-			amount: parseInt( price, 10 ) * newQuantity,
-			currency: productVariations[ productId ].currency,
-		} );
+	// Update BNPL message based on the quantity change
+	quantityInput.on( 'change', ( event ) => {
+		const newQuantity = safeParseInt( event.target.value );
+		const price = safeParseInt( productVariations[ productId ].amount );
+
+		const amount = price * newQuantity;
+		const currency = productVariations[ productId ]?.currency;
+
+		updateBnplPaymentMessage( amount, currency );
 	} );
 
 	// Handle BNPL messaging for variable products.
 	if ( Object.keys( productVariations ).length > 1 ) {
-		$( '.single_variation_wrap' ).on( 'show_variation', function (
-			event,
-			variation
-		) {
-			const quantity = $( '.quantity input[type=number]' ).val();
-			const variationPrice =
-				productVariations[ variation.variation_id ].amount;
-			productId = variation.variation_id;
-			bnplPaymentMessageElement.update( {
-				amount: parseInt( variationPrice, 10 ) * quantity,
-				currency: productVariations[ variation.variation_id ].currency,
-			} );
-		} );
+		// Update BNPL message based on product variation
+		$( '.single_variation_wrap' ).on(
+			'show_variation',
+			( event, variation ) => {
+				if (
+					! quantityInput.length ||
+					! productVariations[ variation.variation_id ]
+				)
+					return;
+
+				const quantity = safeParseInt( quantityInput.val() );
+				const variationPrice = safeParseInt(
+					productVariations[ variation.variation_id ].amount
```

Comment:
Would it make sense using `productVariations[ productId ]?.amount` here?

Reply:
Done https://github.com/Automattic/woocommerce-payments/pull/7153/commits/15d2d7d9d496dd3f5f60de7c92f249cf56e6c348